packages/webapp-libs/webapp-tenants/src/routes/tenantInvitation/tenantInvitation.component.tsx

```diff
@@ -0,0 +1,134 @@
+  const tenants = useTenants();
+  const navigate = useNavigate();
+  const generateTenantPath = useGenerateTenantPath();
+  const generateLocalePath = useGenerateLocalePath();
+  const { toast } = useToast();
+  const intl = useIntl();
+  const { token } = params;
+
+  const acceptSuccessMessage = intl.formatMessage({
+    id: 'Tenant Invitation / Accept / Success message',
+    defaultMessage: '🎉 Invitation accepted!',
+  });
+
+  const declineSuccessMessage = intl.formatMessage({
+    id: 'Tenant Invitation / Decline / Success message',
+    defaultMessage: '🎉 Invitation declined!',
+  });
+
+  const tenant = tenants.find((t) => t?.membership?.invitationToken === token);
+
+  const [commitAcceptMutation, { loading: acceptLoading }] = useMutation(acceptTenantInvitationMutation, {
+    onCompleted: () => {
+      reloadCommonQuery();
+      trackEvent('tenantInvitation', 'accept', tenant?.id);
+      toast({ description: acceptSuccessMessage });
+      if (tenant) navigate(generateTenantPath(RoutesConfig.home, { tenantId: tenant?.id }));
+    },
+  });
+
+  const handleAccept = useCallback(() => {
+    if (!token || !tenant) return;
+    commitAcceptMutation({
+      variables: {
+        input: {
+          token,
+          id: tenant.membership.id,
+        },
+      },
+    });
+  }, [token]);
+
+  const [commitDeclineMutation, { loading: declineLoading }] = useMutation(declineTenantInvitationMutation, {
+    onCompleted: () => {
+      reloadCommonQuery();
+      trackEvent('tenantInvitation', 'decline', tenant?.id);
+      toast({ description: declineSuccessMessage });
+      navigate(generateLocalePath(RoutesConfig.home));
+    },
+  });
+
+  const handleDecline = useCallback(() => {
+    if (!token || !tenant) return;
+    commitDeclineMutation({
+      variables: {
+        input: {
+          token,
+          id: tenant.membership.id,
+        },
+      },
+    });
+  }, [token]);
+
+  let redirectPath: string | null = null;
+
+  if (!tenant) {
+    redirectPath = generateLocalePath(RoutesConfig.home);
+  } else if (tenant.membership?.invitationAccepted) {
+    redirectPath = generateTenantPath(RoutesConfig.home, { tenantId: tenant.id! });
+  }
+
+  useEffect(() => {
+    if (redirectPath) {
+      navigate(redirectPath);
+    }
+  }, [redirectPath]);
+
+  if (redirectPath) {
+    return null;
+  }
+
+  const isLoading = acceptLoading || declineLoading;
+
+  return (
+    <PageLayout>
+      <Card className="max-w-2xl">
+        <CardHeader>
+          <CardTitle>
+            <FormattedMessage defaultMessage="Invitation" id="Tenant Invitation / Page headline" />
+          </CardTitle>
+        </CardHeader>
+        <CardContent>
+          <div>
+            <FormattedMessage
+              defaultMessage="You are invited to join the organization: {organizationName}"
+              id="Tenant Invitation / Invitation message"
+              values={{ organizationName: tenant?.name }}
```

Comment:
I'd wrap this <FormattedMessage /> with the condition `tenant && ...`, otherwise in case of undefined value we can end up with `You are invited to join the organization: undefined`

Reply:
Why was this resolved? I don't see any changes.

Reply:
Sorry... somehow my comment was lost:/ Basically there won't be a situation where the tenant will be null because there is a check above and if the `redirectPath` is falsy, then component will simply return `null`